### PR TITLE
Improve sentry-raven error reporting

### PR DESCRIPTION
**Pull request by mnaichuk**

### CODE REVIEW

File: app/api/v2/exception_handlers.rb [hunk 1]
```diff
@@ -9,19 +9,23 @@ def self.included(base)
...
 
         rescue_from Peatio::Auth::Error do |e|
           report_exception(e)
+          Raven.capture_exception(e) if defined?(Raven)
           error!({ errors: ['jwt.decode_and_verify'] }, 401)
         end
 
-        rescue_from ActiveRecord::RecordNotFound do |_e|
+        rescue_from ActiveRecord::RecordNotFound do |e|
+          Raven.capture_exception(e) if defined?(Raven)
```
Comment (mnaichuk) [inline app/api/v2/exception_handlers.rb]: we can delete it

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -6,6 +6,7 @@
 class ApplicationController < ActionController::Base
   include Authorization::Bearer
   extend Memoist
+  before_action :raven_context
```
Comment (ysv) [inline app/controllers/application_controller.rb]: I would prefer
```
before_action :raven_context, if: -> { defined?(Raven) }
```

File: app/api/v2/auth/jwt_authenticator.rb [hunk 3]
```diff
@@ -21,7 +21,8 @@ def initialize(token)
         # @return [String, Member, NilClass]
         def authenticate
           payload, _header = authenticate!(@token)
-          fetch_member(payload)
+          current_user = fetch_member(payload)
+          raven_context(current_user) if defined?(Raven)
```
Comment (ysv) [inline app/api/v2/auth/jwt_authenticator.rb]: @mnaichuk I would prefer to separate this method from authenticate
Usage:
```ruby
before { authenticate! }
before { set_raven_context! } # find better name
```

**Review by ysv — CHANGES_REQUESTED** [review]

File: app/api/v2/public/mount.rb [hunk 4]
```diff
@@ -5,6 +5,8 @@ module V2
     module Public
       class Mount < Grape::API
 
+        before { set_ets_context! }
```
Comment (ysv) [inline app/api/v2/public/mount.rb:8]: Use another ets_context for public and management API

File: app/api/v2/helpers.rb [hunk 5]
```diff
@@ -10,6 +10,15 @@ def authenticate!
         current_user or raise Peatio::Auth::Error
       end
 
+      def set_ets_context!
+        ::Raven.tags_context(
+          email: current_user.email,
+          uid: current_user.uid,
+          role: current_user.role,
+          Peatio_Version: Peatio::Application::VERSION
```
Comment (ysv) [inline app/api/v2/helpers.rb]: rename to `peatio_version`

**Review by ysv — CHANGES_REQUESTED** [review]

File: app/controllers/application_controller.rb [hunk 6]
```diff
@@ -7,6 +7,8 @@ class ApplicationController < ActionController::Base
   include Authorization::Bearer
   extend Memoist
 
+  before_action :set_raven_context!, if: -> { defined?(Raven) }
```
Comment (ysv) [inline app/controllers/application_controller.rb]: you use `set_raven_context` but method name is `set_ets_context`

**Review by ysv — CHANGES_REQUESTED** [review]

File: app/controllers/application_controller.rb [hunk 7]
```diff
@@ -39,4 +41,13 @@ def not_found
   def is_admin?
     current_user.role.in?(Member::ADMIN_ROLES)
   end
+
+  def set_ets_context!
+    Raven.tags_context(
+      email: current_user.email,
+      uid: current_user.uid,
+      role: current_user.role,
+      Peatio_Version: Peatio::Application::VERSION
```
Comment (ysv) [inline app/controllers/application_controller.rb]: `peatio_version`

**Review by mod — APPROVED** [review]

**Pull request merged**